src/extension.ts

```diff
@@ -30,63 +32,95 @@ import {
   MetalsDidFocus,
   ExecuteClientCommand
 } from "./protocol";
-import { type } from "os";
+import { LazyProgress } from "./lazy-progress";
 
 export async function activate(context: ExtensionContext) {
-  // Make editing Scala docstrings slightly nicer.
-  enableScaladocIndentation();
-
   const userJavaHome = workspace.getConfiguration("metals").get("javaHome");
   if (typeof userJavaHome === "string" && userJavaHome !== "") {
-    startServer(context, userJavaHome);
+    fetchAndLaunchMetals(context, userJavaHome);
   } else {
     require("find-java-home")((err, javaHome) => {
       if (err) window.showErrorMessage("Unable to find Java home.");
-      startServer(context, javaHome);
+      fetchAndLaunchMetals(context, javaHome);
     });
   }
 }
 
-function startServer(context: ExtensionContext, javaHome: string) {
+function fetchAndLaunchMetals(context: ExtensionContext, javaHome: string) {
+  const outputChannel = window.createOutputChannel("Metals");
+  outputChannel.appendLine(`Java home: ${javaHome}`);
+
   const javaPath = path.join(javaHome, "bin", "java");
   const coursierPath = path.join(context.extensionPath, "./coursier");
 
-  const serverVersion = workspace
+  const serverVersion: string = workspace
     .getConfiguration("metals")
     .get("serverVersion");
-
   const serverProperties: string[] = workspace
     .getConfiguration("metals")
     .get("serverProperties")
     .toString()
     .split(" ")
     .filter(e => e.length > 0);
 
-  const javaArgs = serverProperties.concat([
+  const process = spawn(
+    javaPath,
+    serverProperties.concat([
+      "-jar",
+      coursierPath,
+      "fetch",
+      "-p",
+      `org.scalameta:metals_2.12:${serverVersion}`,
+      "-r",
+      "bintray:scalacenter/releases",
+      "-r",
+      "sonatype:releases",
+      "-r",
+      "sonatype:snapshots",
+      "-p"
+    ]),
+    { env: { COURSIER_NO_TERM: "true" } }
+  );
+  const title = `Downloading Metals v${serverVersion}`;
+  trackDownloadProgress(title, outputChannel, process).then(
+    classpath => {
+      launchMetals(
+        outputChannel,
+        context,
+        javaPath,
+        classpath,
+        serverProperties
+      );
+    },
+    _ => {
```

Comment:
`_` still binds a variable, you can simply use `() => {`

For reference, TypeScript suppresses the warning for unused variables if they start with `_`, to allow for uses cases such as `(_dontcare, a) => { /* use a */ }`, so that's why you don't get a diagnostic on this one.

Reply:
👍 Fixed